<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>788</width>
    <height>246</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>QuteSerial</string>
  </property>
  <property name="windowIcon">
   <iconset resource="../resources/icons.qrc">
    <normaloff>:/connect_established.png</normaloff>:/connect_established.png</iconset>
  </property>
  <widget class="QWidget" name="mainArea">
   <layout class="QGridLayout" name="mainGrid">
    <property name="margin">
     <number>2</number>
    </property>
    <property name="spacing">
     <number>1</number>
    </property>
    <item row="0" column="0">
     <widget class="QGroupBox" name="serbox">
      <property name="sizePolicy">
       <sizepolicy hsizetype="MinimumExpanding" vsizetype="Minimum">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="title">
       <string>Serial Port Settings</string>
      </property>
      <property name="flat">
       <bool>true</bool>
      </property>
      <property name="checkable">
       <bool>true</bool>
      </property>
      <layout class="QGridLayout" name="gridLayout">
       <property name="margin">
        <number>2</number>
       </property>
       <property name="spacing">
        <number>1</number>
       </property>
       <item row="0" column="5">
        <widget class="QComboBox" name="baud">
         <property name="currentIndex">
          <number>5</number>
         </property>
         <item>
          <property name="text">
           <string>300</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>600</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>1200</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>2400</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>4800</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>9600</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>19200</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>38400</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>57600</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>115200</string>
          </property>
         </item>
        </widget>
       </item>
       <item row="0" column="6">
        <widget class="QComboBox" name="bits">
         <property name="currentIndex">
          <number>3</number>
         </property>
         <item>
          <property name="text">
           <string>5</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>6</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>7</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>8</string>
          </property>
         </item>
        </widget>
       </item>
       <item row="0" column="1" colspan="4">
        <widget class="QComboBox" name="ports">
         <property name="editable">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="0" column="9">
        <widget class="QComboBox" name="flow">
         <property name="editable">
          <bool>true</bool>
         </property>
         <property name="currentIndex">
          <number>0</number>
         </property>
         <item>
          <property name="text">
           <string>Flow Ctrl: None</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Flow Ctrl: RTS/CTS</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Flow Ctrl: XON/XOFF</string>
          </property>
         </item>
        </widget>
       </item>
       <item row="0" column="7">
        <widget class="QComboBox" name="parity">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="minimumSize">
          <size>
           <width>28</width>
           <height>0</height>
          </size>
         </property>
         <item>
          <property name="text">
           <string extracomment="No Parity">None</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Even</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Odd</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Space</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Mark</string>
          </property>
         </item>
        </widget>
       </item>
       <item row="0" column="8">
        <widget class="QComboBox" name="stopbits">
         <item>
          <property name="text">
           <string>1</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>2</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>1.5</string>
          </property>
         </item>
        </widget>
       </item>
       <item row="2" column="1">
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::Minimum</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>0</width>
           <height>0</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="0" column="12">
        <widget class="QPushButton" name="linebreak">
         <property name="text">
          <string>Break</string>
         </property>
         <property name="icon">
          <iconset resource="../resources/icons.qrc">
           <normaloff>:/red.png</normaloff>
           <normalon>:/green.png</normalon>:/red.png</iconset>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="0" column="10">
        <widget class="QPushButton" name="rts">
         <property name="text">
          <string>RTS</string>
         </property>
         <property name="icon">
          <iconset resource="../resources/icons.qrc">
           <normaloff>:/red.png</normaloff>
           <normalon>:/green.png</normalon>:/red.png</iconset>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
         <property name="flat">
          <bool>false</bool>
         </property>
        </widget>
       </item>
       <item row="0" column="11">
        <widget class="QPushButton" name="dtr">
         <property name="text">
          <string>DTR</string>
         </property>
         <property name="icon">
          <iconset resource="../resources/icons.qrc">
           <normaloff>:/red.png</normaloff>
           <normalon>:/green.png</normalon>:/red.png</iconset>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>788</width>
     <height>19</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionOpen"/>
    <addaction name="actionRescan"/>
    <addaction name="actionStart_Log"/>
    <addaction name="actionClear"/>
   </widget>
   <widget class="QMenu" name="menuMacros">
    <property name="title">
     <string>Macros</string>
    </property>
    <addaction name="editMacros"/>
    <addaction name="separator"/>
    <addaction name="actionAM_1"/>
    <addaction name="actionAM_2"/>
    <addaction name="actionAM_3"/>
    <addaction name="actionAM_4"/>
    <addaction name="actionAM_5"/>
    <addaction name="actionAM_6"/>
    <addaction name="actionAM_7"/>
    <addaction name="actionAM_8"/>
    <addaction name="actionAM_9"/>
    <addaction name="actionAM_10"/>
    <addaction name="actionAM_11"/>
    <addaction name="actionAM_12"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuMacros"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <widget class="QToolBar" name="toolBar">
   <property name="windowTitle">
    <string>toolBar</string>
   </property>
   <property name="iconSize">
    <size>
     <width>32</width>
     <height>32</height>
    </size>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionOpen"/>
   <addaction name="actionRescan"/>
   <addaction name="actionClear"/>
   <addaction name="separator"/>
   <addaction name="editMacros"/>
   <addaction name="actionStart_Log"/>
  </widget>
  <widget class="QToolBar" name="macros">
   <property name="windowTitle">
    <string>toolBar_2</string>
   </property>
   <property name="iconSize">
    <size>
     <width>32</width>
     <height>32</height>
    </size>
   </property>
   <property name="toolButtonStyle">
    <enum>Qt::ToolButtonTextUnderIcon</enum>
   </property>
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionAM_1"/>
   <addaction name="actionAM_2"/>
   <addaction name="actionAM_3"/>
   <addaction name="actionAM_4"/>
   <addaction name="actionAM_5"/>
   <addaction name="actionAM_6"/>
   <addaction name="actionAM_7"/>
   <addaction name="actionAM_8"/>
   <addaction name="actionAM_9"/>
   <addaction name="actionAM_10"/>
   <addaction name="actionAM_11"/>
   <addaction name="actionAM_12"/>
  </widget>
  <action name="actionOpen">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="../resources/icons.qrc">
     <normaloff>:/connect_no.png</normaloff>
     <normalon>:/connect_established.png</normalon>:/connect_no.png</iconset>
   </property>
   <property name="text">
    <string>Connect</string>
   </property>
   <property name="toolTip">
    <string>Open Serial Port</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionRescan">
   <property name="icon">
    <iconset resource="../resources/icons.qrc">
     <normaloff>:/reload.png</normaloff>:/reload.png</iconset>
   </property>
   <property name="text">
    <string>Rescan</string>
   </property>
   <property name="toolTip">
    <string>Rescan for Serial Ports</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+R</string>
   </property>
  </action>
  <action name="editMacros">
   <property name="icon">
    <iconset resource="../resources/icons.qrc">
     <normaloff>:/macro-edit.png</normaloff>:/macro-edit.png</iconset>
   </property>
   <property name="text">
    <string>Edit Macros</string>
   </property>
   <property name="toolTip">
    <string>Edit F1-F12 hotkey macros</string>
   </property>
  </action>
  <action name="actionAM_1">
   <property name="icon">
    <iconset resource="../resources/icons.qrc">
     <normaloff>:/macro.png</normaloff>:/macro.png</iconset>
   </property>
   <property name="text">
    <string/>
   </property>
   <property name="toolTip">
    <string>F1 Macro</string>
   </property>
   <property name="shortcut">
    <string>F1</string>
   </property>
  </action>
  <action name="actionAM_2">
   <property name="icon">
    <iconset resource="../resources/icons.qrc">
     <normaloff>:/macro.png</normaloff>:/macro.png</iconset>
   </property>
   <property name="text">
    <string/>
   </property>
   <property name="toolTip">
    <string>F2 Macro</string>
   </property>
   <property name="shortcut">
    <string>F2</string>
   </property>
  </action>
  <action name="actionAM_3">
   <property name="icon">
    <iconset resource="../resources/icons.qrc">
     <normaloff>:/macro.png</normaloff>:/macro.png</iconset>
   </property>
   <property name="text">
    <string/>
   </property>
   <property name="toolTip">
    <string>F3 Macro</string>
   </property>
   <property name="shortcut">
    <string>F3</string>
   </property>
  </action>
  <action name="actionAM_4">
   <property name="icon">
    <iconset resource="../resources/icons.qrc">
     <normaloff>:/macro.png</normaloff>:/macro.png</iconset>
   </property>
   <property name="text">
    <string/>
   </property>
   <property name="toolTip">
    <string>F4 Macro</string>
   </property>
   <property name="shortcut">
    <string>F4</string>
   </property>
  </action>
  <action name="actionAM_5">
   <property name="icon">
    <iconset resource="../resources/icons.qrc">
     <normaloff>:/macro.png</normaloff>:/macro.png</iconset>
   </property>
   <property name="text">
    <string/>
   </property>
   <property name="toolTip">
    <string>F5 Macro</string>
   </property>
   <property name="shortcut">
    <string>F5</string>
   </property>
  </action>
  <action name="actionAM_6">
   <property name="icon">
    <iconset resource="../resources/icons.qrc">
     <normaloff>:/macro.png</normaloff>:/macro.png</iconset>
   </property>
   <property name="text">
    <string/>
   </property>
   <property name="toolTip">
    <string>F6 Macro</string>
   </property>
   <property name="shortcut">
    <string>F6</string>
   </property>
  </action>
  <action name="actionAM_7">
   <property name="icon">
    <iconset resource="../resources/icons.qrc">
     <normaloff>:/macro.png</normaloff>:/macro.png</iconset>
   </property>
   <property name="text">
    <string/>
   </property>
   <property name="toolTip">
    <string>F7 Macro</string>
   </property>
   <property name="shortcut">
    <string>F7</string>
   </property>
  </action>
  <action name="actionAM_8">
   <property name="icon">
    <iconset resource="../resources/icons.qrc">
     <normaloff>:/macro.png</normaloff>:/macro.png</iconset>
   </property>
   <property name="text">
    <string/>
   </property>
   <property name="toolTip">
    <string>F8 Macro</string>
   </property>
   <property name="shortcut">
    <string>F8</string>
   </property>
  </action>
  <action name="actionAM_9">
   <property name="icon">
    <iconset resource="../resources/icons.qrc">
     <normaloff>:/macro.png</normaloff>:/macro.png</iconset>
   </property>
   <property name="text">
    <string/>
   </property>
   <property name="toolTip">
    <string>F9 Macro</string>
   </property>
   <property name="shortcut">
    <string>F9</string>
   </property>
  </action>
  <action name="actionAM_10">
   <property name="icon">
    <iconset resource="../resources/icons.qrc">
     <normaloff>:/macro.png</normaloff>:/macro.png</iconset>
   </property>
   <property name="text">
    <string/>
   </property>
   <property name="toolTip">
    <string>F10 Macro</string>
   </property>
   <property name="shortcut">
    <string>F10</string>
   </property>
  </action>
  <action name="actionAM_11">
   <property name="icon">
    <iconset resource="../resources/icons.qrc">
     <normaloff>:/macro.png</normaloff>:/macro.png</iconset>
   </property>
   <property name="text">
    <string/>
   </property>
   <property name="toolTip">
    <string>F11 Macro</string>
   </property>
   <property name="shortcut">
    <string>F11</string>
   </property>
  </action>
  <action name="actionAM_12">
   <property name="icon">
    <iconset resource="../resources/icons.qrc">
     <normaloff>:/macro.png</normaloff>:/macro.png</iconset>
   </property>
   <property name="text">
    <string/>
   </property>
   <property name="toolTip">
    <string>F12 Macro</string>
   </property>
   <property name="shortcut">
    <string>F12</string>
   </property>
  </action>
  <action name="actionStart_Log">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="../resources/icons.qrc">
     <normaloff>:/log.png</normaloff>:/log.png</iconset>
   </property>
   <property name="text">
    <string>Start Log</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+L</string>
   </property>
  </action>
  <action name="actionClear">
   <property name="icon">
    <iconset resource="../resources/icons.qrc">
     <normaloff>:/clear.png</normaloff>:/clear.png</iconset>
   </property>
   <property name="text">
    <string>Clear</string>
   </property>
   <property name="toolTip">
    <string>Clears Visible Transmit and Receive Logs</string>
   </property>
   <property name="shortcut">
    <string>Alt+W</string>
   </property>
  </action>
 </widget>
 <tabstops>
  <tabstop>ports</tabstop>
  <tabstop>serbox</tabstop>
  <tabstop>baud</tabstop>
  <tabstop>bits</tabstop>
  <tabstop>parity</tabstop>
  <tabstop>stopbits</tabstop>
  <tabstop>flow</tabstop>
  <tabstop>rts</tabstop>
  <tabstop>dtr</tabstop>
 </tabstops>
 <resources>
  <include location="../resources/icons.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>serbox</sender>
   <signal>toggled(bool)</signal>
   <receiver>baud</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>383</x>
     <y>231</y>
    </hint>
    <hint type="destinationlabel">
     <x>293</x>
     <y>237</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>serbox</sender>
   <signal>toggled(bool)</signal>
   <receiver>ports</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>383</x>
     <y>231</y>
    </hint>
    <hint type="destinationlabel">
     <x>128</x>
     <y>237</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>serbox</sender>
   <signal>toggled(bool)</signal>
   <receiver>bits</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>383</x>
     <y>231</y>
    </hint>
    <hint type="destinationlabel">
     <x>365</x>
     <y>237</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>serbox</sender>
   <signal>toggled(bool)</signal>
   <receiver>parity</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>383</x>
     <y>231</y>
    </hint>
    <hint type="destinationlabel">
     <x>428</x>
     <y>237</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>serbox</sender>
   <signal>toggled(bool)</signal>
   <receiver>stopbits</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>383</x>
     <y>231</y>
    </hint>
    <hint type="destinationlabel">
     <x>491</x>
     <y>237</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>serbox</sender>
   <signal>toggled(bool)</signal>
   <receiver>flow</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>383</x>
     <y>231</y>
    </hint>
    <hint type="destinationlabel">
     <x>684</x>
     <y>237</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>serbox</sender>
   <signal>toggled(bool)</signal>
   <receiver>rts</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>383</x>
     <y>231</y>
    </hint>
    <hint type="destinationlabel">
     <x>650</x>
     <y>109</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>serbox</sender>
   <signal>toggled(bool)</signal>
   <receiver>dtr</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>383</x>
     <y>231</y>
    </hint>
    <hint type="destinationlabel">
     <x>726</x>
     <y>109</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionOpen</sender>
   <signal>toggled(bool)</signal>
   <receiver>ports</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>78</x>
     <y>109</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>serbox</sender>
   <signal>toggled(bool)</signal>
   <receiver>linebreak</receiver>
   <slot>setVisible(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>393</x>
     <y>152</y>
    </hint>
    <hint type="destinationlabel">
     <x>746</x>
     <y>109</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionOpen</sender>
   <signal>toggled(bool)</signal>
   <receiver>actionRescan</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
